libvuln/jsonblob/jsonblob.go

```diff
@@ -45,85 +45,43 @@ type Store struct {
 }
 
 // Load reads in all the records serialized in the provided [io.Reader].
-func Load(ctx context.Context, r io.Reader) (*Loader, error) {
-	l := Loader{
-		dec: json.NewDecoder(r),
-		cur: uuid.Nil,
+func Load(ctx context.Context, r io.Reader) (*Store, error) {
+	s, err := New()
+	if err != nil {
+		return nil, err
 	}
-	return &l, nil
-}
-
-// Loader is an iterator that returns a series of [Entry].
-//
-// Users should call [*Loader.Next] until it reports false, then check for
-// errors via [*Loader.Err].
-type Loader struct {
-	err error
-	e   *Entry
-
-	dec  *json.Decoder
-	next *Entry
-	de   diskEntry
-	cur  uuid.UUID
-}
 
-// Next reports whether there's an [Entry] to be processed.
-func (l *Loader) Next() bool {
-	if l.err != nil {
-		return false
+	l, err := NewLoader(r)
+	if err != nil {
+		return nil, err
 	}
 
-	for l.err = l.dec.Decode(&l.de); l.err == nil; l.err = l.dec.Decode(&l.de) {
-		id := l.de.Ref
-		// If we just hit a new Entry, promote the current one.
-		if id != l.cur {
-			l.e = l.next
-			l.next = &Entry{}
-			l.next.Updater = l.de.Updater
-			l.next.Fingerprint = l.de.Fingerprint
-			l.next.Date = l.de.Date
+	// TODO(DO NOT MERGE): ~~This implementation might be a bit naive. Currently,
+	//  it basically copies [OfflineImport]. We could probably do some custom
+	//  parsing such that it basically just decodes the json into the
+	//  appropriate [Store] fields.~~
+	//  Actually, this might be the way.
+	//  [OfflineImport]: https://github.com/quay/claircore/blob/126f688bb11220fb34708719be91952dc32ff7b1/libvuln/updates.go#L17-L74
```

Comment:
Couple of other thoughts here - we could either:
1. Change the documentation to be specific to `Loader` instead of `Store`
2. Merge https://github.com/quay/claircore/pull/1334, then use `Unmarshal()` instead of using a `Loader` to load the data.